mesh/mesh-io-tcpserver.c

```diff
@@ -388,35 +416,53 @@ static bool tls_ctx_init(struct mesh_io_private *pvt)
 	return true;
 }
 
-static bool process_tls_args(char *argv[],
-				size_t argc, struct mesh_io_private *pvt)
+static bool acl_entry_added(struct tcpserver_acl *acl, uint8_t *identity_buf,
+								void *user_data)
 {
-	size_t len;
+	char			identity[IDENTITY_LEN];
+	struct mesh_io		*mesh_io = user_data;
+	struct tcpserver_conn	*conn = tcpserver_conn_new(identity_buf);
 
-	pvt->uuid = l_util_from_hexstring(argv[1], &len);
+	if (!l_queue_push_tail(mesh_io->pvt->conns, conn))
+		return false;
 
-	if (!pvt->uuid || len != UUID_LEN) {
-		l_error("Invalid UUID length");
+	if (!hex2str(identity_buf, IDENTITY_BUF_LEN, identity,
+							     sizeof(identity)))
 		return false;
-	}
 
-	pvt->dev_key = l_util_from_hexstring(argv[2], &len);
+	conn->conn_stat = conn_stat_new(mesh_io->pvt->dbus_path,
+					mesh_io->pvt->dbus, identity);
+	return true;
+}
 
-	if (!pvt->dev_key || len != DEV_KEY_LEN) {
-		l_error("Invalid 'dev-key' length");
-		return false;
-	}
+static bool acl_entry_loaded(struct tcpserver_acl *acl, uint8_t *identity_buf,
+								void *user_data)
+{
+	struct mesh_io		*mesh_io = user_data;
+	struct tcpserver_conn	*conn = tcpserver_conn_new(identity_buf);
+
+	return l_queue_push_tail(mesh_io->pvt->conns, conn);
+}
 
-	pvt->net_key = l_util_from_hexstring(argv[3], &len);
+static bool acl_entry_removed(struct tcpserver_acl *acl, uint8_t *identity_buf,
+								void *user_data)
+{
+	struct mesh_io		*mesh_io = user_data;
+	struct tcpserver_conn	*conn;
 
-	if (!pvt->net_key || len != NET_KEY_LEN) {
-		l_error("Invalid 'net-key' length");
+	conn = l_queue_find(mesh_io->pvt->conns,
+			match_tcpserver_conn_identity_buf, identity_buf);
+	if (!conn)
 		return false;
-	}
 
-	return true;
+	tcpserver_conn_destroy(conn);
+
+	// TODO(prfi):
+	l_queue_remove(mesh_io->pvt->conns, conn);
```

Comment:
Why is this a `TODO`? Looks fine?

Reply:
There is `return false;` to avoid overwriting configuration file (just for test purposes).